compiler-core/src/type_/error.rs

```diff
@@ -841,6 +841,7 @@ pub enum FeatureKind {
     NestedTupleAccess,
     InternalAnnotation,
     AtInJavascriptModules,
+    TypeNarrowing,
```

Comment:
I'd define two different variants to produce nicer messages:
- one to talk about narrowing for safe record updates
- one to talk about narrowing for safe accessors

Reply:
Good idea!